### PR TITLE
Feature/custom esbuild

### PR DESCRIPTION
**Pull request by sandeep-vedam**
Added support for custom esbuild plugin

### CODE REVIEW

File: src/configs/esbuild.es5.config.js [hunk 1]
```diff
@@ -28,6 +28,13 @@ const babelPluginTransFormSpread = require('@babel/plugin-transform-spread')
 const babelPluginTransFormParameters = require('@babel/plugin-transform-parameters')
 const babelPluginClassProperties = require('@babel/plugin-proposal-class-properties')
 const babelPluginInlineJsonImport = require('babel-plugin-inline-json-import')
+const deepMerge = require('deepmerge')
+
+let customConfig
+
+if (process.env.LNG_CUSTOM_ESBUILD === 'true') {
+  customConfig = require(path.join(process.cwd(), 'esbuild.es6.config'))
```
Comment (ricardo-a-alves-alb) [inline src/configs/esbuild.es5.config.js:36]: Should be es5. And what about the _env_?

File: src/configs/esbuild.es6.config.js [hunk 2]
```diff
@@ -26,6 +26,13 @@ const babelPresetEnv = require('@babel/preset-env')
 const path = require('path')
 const babelPluginClassProperties = require('@babel/plugin-proposal-class-properties')
 const babelPluginInlineJsonImport = require('babel-plugin-inline-json-import')
+const deepMerge = require('deepmerge')
+
+let customConfig
+
+if (process.env.LNG_CUSTOM_ESBUILD === 'true') {
+  customConfig = require(path.join(process.cwd(), 'esbuild.es6.config'))
```
Comment (ricardo-a-alves-alb) [inline src/configs/esbuild.es6.config.js:34]: What about the _env_?

**Review by ricardo-a-alves-alb — CHANGES_REQUESTED** [review]

File: src/configs/esbuild.es6.config.js (hunk 2 above)
Comment (sandeep-vedam) [reply src/configs/esbuild.es6.config.js:34]: Hey @ricardo-a-alves-alb ,
The ones that are mentioned in the env will be ignored here and the custom config takes the priority here.

Comment (ricardo-a-alves-alb) [reply src/configs/esbuild.es6.config.js:34]: I've misread the configuration description. By `env`, I thought it meant like `development`, `staging` and `production`. But now I see that it's about `es5` and `es6`.

**Review by michielvandergeest — APPROVED** [review]
LGTM

**Pull request merged**